### DISCUSSION TITLE
Lexer does not invoke walkTokens?

### DISCUSSION THREAD

**Discussion by ivanjaros** [markedjs/marked#3414] [Q&A] [answered]
When I call `mdInstance.lexer(value)` my `walkTokens` added via `mdInstance.use()` are not being triggered. Is there some special way to force Lexer to call them? Even if I add walkTokens as `new Marked({walkTokens: (token) => {}})`, it is ignored by the Lexer.

**Comment by UziTech** [comment, marked as answer]
If you are not using `mdInstance.parse` you will need to call me `mdInstance.walkTokens` by yourself after the Lexer creates the tokens. See https://github.com/markedjs/marked/blob/master/src/Instance.ts#L300

**Reply by ivanjaros** [reply]
Good to know. Would be better if it would be documented to avoid surprises but I guess not many people only use the lexer. Luckily, I managed to do what I needed without the function, this time.

**Reply by UziTech** [reply]
If you would like to create a PR to update the docs we could get it in there 😁👍

**Reply by ivanjaros** [reply]
I don't know this library enough to confidently make a PR, especially for documentation, merely on your good word. Sorry.

**Reply by UziTech** [reply]
Neither did I the first time I created a PR
